File: AN_2/SEM_2/MPP/c#/lab3/domain/ParticipantCursa.cs
namespace lab3.domain;

public class ParticipantCursa
{
    private Participant part;
    private Cursa cursa;

    public ParticipantCursa(Participant part, Cursa cursa)
    {
        this.part = part;
        this.cursa = cursa;
    }

    public Participant GetParticipant() => part;
    public void SetParticipant(Participant part) => this.part = part;

    public Cursa GetCursa() => cursa;
    public void SetCursa(Cursa cursa) => this.cursa = cursa;

    public override bool Equals(object? obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;
        var that = (ParticipantCursa)obj;
        return Equals(part, that.part) && Equals(cursa, that.cursa);
    }

    public override int GetHashCode() => HashCode.Combine(part, cursa);

    public override string ToString()
    {
        return $"ParticipantCursa{{part={part}, cursa={cursa}}}";
    }
}

File: AN_2/SEM_2/MPP/java/lab6/Model/src/main/java/ro/mpp2024/model/ParticipantCursa.java
package ro.mpp2024.model;

import java.io.Serializable;
import java.util.Objects;

public class ParticipantCursa implements Serializable {
    private Participant participant;
    private Cursa cursa;

    public ParticipantCursa(Participant participant, Cursa cursa) {
        this.participant = participant;
        this.cursa = cursa;
    }

    public Participant getParticipant() {
        return participant;
    }

    public void setParticipant(Participant participant) {
        this.participant = participant;
    }

    public Cursa getCursa() {
        return cursa;
    }

    public void setCursa(Cursa cursa) {
        this.cursa = cursa;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ParticipantCursa that = (ParticipantCursa) o;
        return Objects.equals(participant, that.participant) && Objects.equals(cursa, that.cursa);
    }

    @Override
    public int hashCode() {
        return Objects.hash(participant, cursa);
    }

    @Override
    public String toString() {
        return "ParticipantCursa{" +
                "participant=" + participant +
                ", cursa=" + cursa +
                '}';
    }
}
